src/goto-programs/class_hierarchy.cpp

```diff
@@ -30,6 +31,9 @@ void class_hierarchyt::operator()(const symbol_tablet &symbol_table)
     {
       const struct_typet &struct_type = to_struct_type(symbol_pair.second.type);
 
+      class_map[symbol_pair.first].is_abstract =
+        struct_type.get_bool(ID_abstract);
+
```

Comment:
Feels like the kind of thing that could do with a test case.

Reply:
Not sure what you want me to test here. Actually, that output has already proved useful to check whether we load interface/abstract class information correctly.

Reply:
Just a test that uses that part.  I ask because the code was right / finished before and now this needs to be added so I think we should have a case that was sub-par / wrong / insufficient before and is now better.  If it's a pain don't worry.